### ISSUE TITLE
Google ReCaptcha may not be checking if captcha is completed

### ISSUE THREAD

**Issue by BenParizek** [barrelstrength/craft-sprout-forms-google-recaptcha#10]
User reports captcha having no effect on spam

<img src="https://frontapp.com/assets/img/favicons/favicon-32x32.png" height="16" width="16" alt="Front logo" /> [Front conversations](https://app.frontapp.com/open/top_12b9f)

**Comment by BenParizek** [comment]
This was a configuration issue.

**Comment by rbrv** [comment]
@BenParizek Can you please elaborate? We're having the same problem...

**Comment by BenParizek** [comment]
@rbrv We'll be releasing an update soon to try to address some of this but here are some things to consider:

- Currently, this Captcha doesn't make the reCAPTCHA field required if a user does not fill it out. You will need to add the code below to the template to help make that happen. We'll be adding an option to output that code in the settings in the next release. We had previously removed it because another user reported it conflicting with other things they were doing.

``` twig
{% css %}
#g-recaptcha-response {
    display: block !important;
    position: absolute;
    margin: -78px 0 0 0 !important;
    width: 302px !important;
    height: 76px !important;
    z-index: -999999;
    opacity: 0;
}
{% endcss %}
```

Also note that reCAPTCHA has settings on the Google side where you can increase the **Security Preference**: https://developers.google.com/recaptcha/docs/settings

If those items still don't help solve things for you, let us know. The next release of Sprout Forms will have improved Spam logging to help diagnose more about what characterizes your spam traffic.

**Comment by rbrv** [comment]
As we weren't sure when there would be a fix, we've had to switch to WheelForm for this project. 
Will try again on our next project. Thanks @BenParizek!

**Comment by BenParizek** [comment]
@rbrv fyi, yesterday we released **Google ReCaptcha For Sprout Forms v1.0.6** which adds the option to enable/disable the output the CSS above and make the field required on the front-end. 

We also released **Sprout Forms v3.6.0** which adds spam management features including flagging entries that fail a Captcha check with a spam status in the database and filtering them out of the main listing views, additional tracking metadata like IP Address and a new Spam Log report to view data on what Captchas have failed and why.
